Components/Helper.php

```diff
@@ -878,4 +894,86 @@ public function getProductCountForCategoryRecovery()
             SELECT COUNT(*) FROM `s_plugin_connect_items` WHERE shop_id IS NOT NULL GROUP BY `article_id`
           ) AS Z')->fetchColumn();
     }
+
+    /**
+     * adds the shopId to the ConnectCategoriesTable
+     * @param int $offset
+     * @param int $batchsize
+     */
+    public function addShopIdToConnectCategories($offset, $batchsize)
+    {
+        $result = $this->manager->getConnection()->executeQuery('SELECT `article_id`, `category`, `shop_id` FROM `s_plugin_connect_items` WHERE shop_id IS NOT NULL GROUP BY `article_id` ORDER BY `id` LIMIT ? OFFSET ?',
+            [$batchsize, $offset],
+            [\PDO::PARAM_INT, \PDO::PARAM_INT]
+        );
+
+        while ($row = $result->fetch()) {
+            $categories = json_decode($row['category'], true);
+            $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_product_to_categories WHERE articleID = ?',
+                [$row['article_id']]
+            );
+
+            $shopId = (int) $row['shop_id'];
+            foreach ($categories as $categoryKey => $category) {
+                $selectedCategory = $this->manager->getConnection()->executeQuery('SELECT `id` FROM s_plugin_connect_categories WHERE category_key = ? AND shop_id = ?',
```

Comment:
no i can't simply do that in one query because first i check if that connect category with that specific shopId exists
if this is the case i don't need to update ConnectCategories
in the second query with shopId = null i check for the "old" categories without shopId then i can simply update that row and add the shopId so i don't need to insert a new row

but nevertheless i extracted the code to make it more readable